src/main/java/org/commonjava/util/partyline/FileTree.java

```diff
@@ -327,46 +327,55 @@ boolean tryLock( File file, String ownerName, String label, LockLevel lockLevel,
         while ( end < 1 || System.currentTimeMillis() < end )
         {
             T result = tryLock( realFile, Thread.currentThread().getName(), "Open File for " + ( doOutput ? "output" : "input" ),
-                                           doOutput ? LockLevel.write : read, WAIT_TIMEOUT, unit, ( opLock ) -> {
+                                           doOutput ? LockLevel.write : read, timeout, unit, ( opLock ) -> {
                         FileEntry entry = entryMap.get( realFile.getAbsolutePath() );
-                        if ( entry.file != null )
+                        boolean proceed = false;
+                        synchronized ( entry )
```

Comment:
Agreed, regarding the synchronized block around the entryMap / fileEntry stuff. *sigh*

There are two things going on WRT locking in partyline:

* long-term, non-exclusive usage locks on files...allowing particular mixtures of reads/writes and exclusive deletes
* operation locks. These are like synchronized keywords on methods, except I'm trying to reduce the scope of the lock to the file path in question (to avoid locking the whole system for each file open / close / delete operation).

The operation locks are in a WeakHashMap to allow them to be GC'ed when the string path is GC'ed...which (*I think*) shouldn't be immediately after the operation completes...but if it is, that might be okay too, since it's really trying to prevent concurrent setup of multiple FileEntry / LockOwner instances for the same file.

I'm working on a better way to handle this, and I think I might have figured out something...but I need to check it on my localhost a bit more before I push an update.